### ISSUE TITLE
RISC-V vDSO can't be linked with LLD: no input files

### ISSUE THREAD

**Issue by tpimh** [ClangBuiltLinux/linux#805]
For BFD the following works: `riscv64-linux-gnu-ld -melf64lriscv -r -R arch/riscv/kernel/vdso/vdso-dummy.o -o arch/riscv/kernel/vdso/vdso-syms.o`
For LLD it fails:
```
   ld.lld-10  -melf64lriscv   -r -R arch/riscv/kernel/vdso/vdso-dummy.o -o arch/riscv/kernel/vdso/vdso-syms.o
ld.lld-10: error: no input files
```

**Comment by nickdesaulniers** [comment]
were those `.o` files created?

**Comment by MaskRay** [comment]
Can you share a RISC-V build command line?

**Comment by nathanchance** [comment]
```
# Need linux-next for RISCV right now
$ git clone git://git.kernel.org/pub/scm/linux/kernel/git/next/linux-next.git

$ cd linux-next

$ make -j$(nproc) -s ARCH=riscv AR=llvm-ar AS=clang CC=clang CROSS_COMPILE=riscv64-linux-gnu- LD=ld.lld OBJCOPY=llvm-objcopy OBJDUMP=llvm-objdump STRIP=llvm-strip O=out/riscv64 distclean defconfig arch/riscv/kernel/vdso/
```

should be enough to reproduce this.

**Comment by ihalip** [comment]
Could it be that ld.lld aliases `-R` to `-rpath` instead of `--just-symbols`? I can see that using `--just-symbols` instead of `-R`, I get some valid output (in the sense that lld includes the symbols output by ld, and a handful of others).

**Comment by nickdesaulniers** [comment]
accepted: https://git.kernel.org/pub/scm/linux/kernel/git/riscv/linux.git/commit/?h=fixes&id=3c1918c8f54166598195d938564072664a8275b1

**Comment by nathanchance** [comment]
Merged into mainline: https://git.kernel.org/linus/3c1918c8f54166598195d938564072664a8275b1
